web/packages/design/src/utils/match/matchers.test.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { dateTimeMatcher, dateMatcher } from './matchers';
+
+test('dateTimeMatcher should match date times correctly', () => {
+  const searchValue = '23:13';
+
+  const dateMatched = new Date('2023-05-04T23:13:55.539Z');
+  const dateMatchedISOString = '2023-05-04T23:13:55.539Z';
+  const dateNoMatch = new Date('2023-05-04T12:11:00.539Z');
```

Comment:
we should also test for invalid date eg `new Date('invalid')`, if we have an invalid date, make it return false